cmd/server/helper_keys.go

```diff
@@ -18,7 +18,7 @@ func createRS256KeysIfNotExist(c *config.Config, set, kid, use string) {
 	if _, err := ctx.KeyManager.GetKey(set, kid); errors.Cause(err) == pkg.ErrNotFound {
 		logrus.Infof("Key pair for signing %s is missing. Creating new one.", set)
 
-		keys, err := generator.Generate("")
+		keys, err := generator.Generate(set)
```

Comment:
The API contract would remain the same, but there may need to be some changes to `/keys/set` on the backend. 

Google:   
https://www.googleapis.com/oauth2/v3/certs   
MS:   
https://login.windows.net/common/discovery/keys  

They are using just a uuid for the `kid`, so thats another option.